app/components/progress_component/progress_component_controller.js

```diff
@@ -1,14 +1,15 @@
 import ApplicationController from "../../webpacker/javascript/controllers/application_controller"
 
 export default class ProgressComponentController extends ApplicationController {
-  static targets = ["counter", "back", "continue", "submit", "step"]
+  static targets = ["counter", "back", "continue", "submit", "step", "warning"]
 
   connect() {
     if (this.stepTargets.length == 0) throw new Error("Must have at least one step")
     this.currentStep = 0
     this.updateCounter()
```

Comment:
Could we just call updateSection in the connect method, rather than having this long list of methods?